internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -47,6 +48,206 @@ func New(sess *session.Session) CloudFormation {
 	}
 }
 
+// CreateProjectResources sets up everything required for our project-wide resources.
+// These resources include things that are regional, rather than scoped to a particular
+// environment, such as ECR Repos, CodePipeline KMS keys & S3 buckets.
+// We deploy project resources through StackSets - that way we can have one
+// template that we update and all regional stacks are updated.
+func (cf CloudFormation) CreateProjectResources(project *archer.Project) error {
+	projectConfig := newProjectStackConfig(project, cf.box)
+
+	// First deploy the project roles needed by StackSets. These roles
+	// allow the stack set to set up our regional stacks.
+	if err := cf.deploy(projectConfig); err == nil {
+		_, err := cf.waitForStackCreation(projectConfig)
+		if err != nil {
+			return err
+		}
+	} else {
+		// If the stack already exists - we can move on
+		// to creating the StackSet.
+		var alreadyExists *ErrStackAlreadyExists
+		if !errors.As(err, &alreadyExists) {
+			return err
+		}
+	}
+
+	blankProjectTemplate, err := projectConfig.ResourceTemplate(&projectResourcesConfig{
+		Project: projectConfig.Project.Name,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	_, err = cf.client.CreateStackSet(&cloudformation.CreateStackSetInput{
+		Description:           aws.String(projectConfig.StackSetDescription()),
+		StackSetName:          aws.String(projectConfig.StackSetName()),
+		TemplateBody:          aws.String(blankProjectTemplate),
+		ExecutionRoleName:     aws.String(projectConfig.StackSetExecutionRoleName()),
+		AdministrationRoleARN: aws.String(projectConfig.StackSetAdminRoleARN()),
+		Tags:                  projectConfig.Tags(),
+	})
+
+	if err != nil && !stackSetExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateProject updates Project resources in all regions. If a new app or env is added
+func (cf CloudFormation) UpdateProject(project *archer.Project, newApps []*archer.Application, newEnvs []*archer.Environment) error {
+	projectConfig := newProjectStackConfig(project, cf.box)
+
+	if err := cf.updateStackSet(projectConfig, newApps, newEnvs); err != nil {
+		return err
+	}
+	return cf.addNewStackInstances(projectConfig, newEnvs)
+}
+
+// updateStackSet generates a template with several project level resources:
+// * ECR Repos for every app and appropriate cross-account resource policies
+// * S3 Buckets & KMS Keys for CodePipeline
+// It looks at the previously deployed stack, and appends the apps and account IDs
+// needed for those resources (ensuring we don't accidentally delete any resources).
+// We do this, instead of reading from the project store because the project store could
+// be eventually consistent, meaning we might miss an app or environment in certain edge cases.
+// While this consistency is fine for purely UX situations, it's not ok when we're generating
+// resources.
+func (cf CloudFormation) updateStackSet(projectConfig *projectStackConfig, apps []*archer.Application, envs []*archer.Environment) error {
```

Comment:
With the name of this method, I'd have expected for it to be agnostic of our concepts (that's how I've been treating private methods in the `cloudformation` package)

I think it's fine to break that but can we rename to `updateProjectWithNewApps` or something like that? to denote it's project specific